### ISSUE TITLE
PowerCLI module is broken in PS 7.2 due to a breaking change in Newtonsoft.Json v13.0.1

### ISSUE THREAD

**Issue by daxian-dbw** [vmware/PowerCLI-Example-Scripts#494]
### Steps to reproduce

PowerShell 7.2.0 (The error does not appear with PowerShell 7.0 LTS or with 7.1.5.)
Install VMware PowerCLI
Import Module: https://github.com/vmware/PowerCLI-Example-Scripts/blob/master/Modules/Backup-VCSA/**Backup-VCSA.psm1**
Connect-CisServer
Get-VCSABackupJobs

### Expected behavior

```console
Receive information from PowerCLI API.

The error does not appear with PowerShell 7.0 LTS or with 7.1.5.
```

### Actual behavior

```console
The reader's MaxDepth of 64 has been exceeded.
```

### Error details

```console
Get-CisService: 12.11.2021 17:19:49 Get-CisService  One or more errors occurred. (One or more errors occurred. (The reader's MaxDepth of 64 has been exceeded. Path
'result.output.STRUCTURE['com.vmware.vapi.metadata.metamodel.component_data'].info.STRUCTURE['com.vmware.vapi.metadata.metamodel.component_info'].packages[7].STRUCTURE.map-entry.value.STRUCTURE['com.vmware.vapi.metadata.metamodel.packa
ge_info'].services[0].STRUCTURE.map-entry.value.STRUCTURE['com.vmware.vapi.metadata.metamodel.service_info'].structures[6].STRUCTURE.map-entry.value.STRUCTURE['com.vmware.vapi.metadata.metamodel.structure_info'].fields[5].STRUCTURE['co
m.vmware.vapi.metadata.metamodel.field_info'].type.STRUCTURE['com.vmware.vapi.metadata.metamodel.type'].generic_instantiation.OPTIONAL.STRUCTURE['com.vmware.vapi.metadata.metamodel.generic_instantiation'].element_type.OPTIONAL.STRUCTUR
E['com.vmware.vapi.metadata.metamodel.type'].generic_instantiation.OPTIONAL.STRUCTURE['com.vmware.vapi.metadata.metamodel.generic_instantiation'].map_value_type.OPTIONAL.STRUCTURE['com.vmware.vapi.metadata.metamodel.type'].generic_inst
antiation.OPTIONAL.STRUCTURE['com.vmware.vapi.metadata.metamodel.generic_instantiation'].map_value_type.OPTIONAL.STRUCTURE['com.vmware.vapi.metadata.metamodel.type'].generic_instantiation.OPTIONAL.STRUCTURE['com.vmware.vapi.metadata.me
tamodel.generic_instantiation'].element_type', line 1, position 567214.))

Get-CisService: 12.11.2021 17:19:49 Get-CisService  CisService with name 'com.vmware.appliance.recovery.backup.job' was not found using the specified filter(s). 

Write-Error: Cannot bind argument to parameter 'Exception' because it is null.
```

### Environment data

```powershell
Debian 10 amd64

PS /usr/lib/zabbix/externalscripts> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.2.0
PSEdition                      Core
GitCommitId                    7.2.0
OS                             Linux 4.19.0-16-amd64 #1 SMP Debian 4.19.181-1 (2021-03-19)
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by daxian-dbw** [comment]
Quoted from: https://stackoverflow.com/a/68580426:
> In Json.NET [13.0.1](https://github.com/JamesNK/Newtonsoft.Json/releases/tag/13.0.1), Newtonsoft changed the default MaxDepth for JsonReader (and hence all derived reader types including JsonTextReader) to 64:
>> Change - JsonReader and JsonSerializer MaxDepth defaults to 64

The `ConvertFrom-Json` command in PowerShell already specified the `MaxDepth` in `JsonSerializerSettings`, so it should not be affected. Therefore, it's likely that the PowerCLI module is directly calling APIs from `Newtonsoft.Json.dll`, and should be updated accordingly. @xenadmin can you please open an issue in https://github.com/vmware/PowerCLI-Example-Scripts about this?

https://github.com/PowerShell/PowerShell/blob/7dc4587014bfa22919c933607bf564f0ba53db2e/src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs#L179-L187

**Comment by xenadmin** [comment]
Additionally I opened an official VMware support request (21277890011) as per: https://blogs.vmware.com/PowerCLI/2017/08/powercli-support-breakdown.html

**Comment by Alta2048** [comment]
Hi @xenadmin , thanks for reporting this . Do you know if VMware Support provided you with an Update/ETA on this issue ? It also affect  function [Get-CisService](https://developer.vmware.com/docs/powercli/latest/vmware.vimautomation.cis.core/commands/get-cisservice/#Default)

**Comment by xenadmin** [comment]
I got a phone call from VMware the next day, where they acknowledged the issue, informed me that they will immediately close the case, as this will be tracked here in this Github issue, and told me to watch out for newer releases of PowerCLI.
To workaround this this issue, I had to downgrade to PowerShell LTS in the meantime.

**Comment by kamennikolov** [comment]
I can confirm that we're aware of the issue and we're working to address it in the next PowerCLI version.

**Comment by mcarpendale** [comment]
@kamennikolov - can you please share the workaround until the new version of PowerCLI is released?

**Comment by kamennikolov** [comment]
Currently the only workaround is to use an older PowerShell version - 7.0 or 7.1

**Comment by xenadmin** [comment]
@mcarpendale In my use-case, I have Debian (10) VMs, with powershell and powercli, where my scripts fail. On Debian I had to downgrade to `powershell-lts` to make everything work again, which looks like this as of today:
![grafik](https://user-images.githubusercontent.com/12057348/145171502-ce366a75-8ba4-461a-bad0-ee8fe2a06d78.png)

**Comment by mcarpendale** [comment]
thanks @kamennikolov and @xenadmin - 7.1.5 is working from me again

**Comment by MallocArray** [comment]
Confirmed still not working with Powershell 7.2.1 and PowerCLI 12.4.1
Using Windows Powershell 5.1 is fine

**Comment by abidlen** [comment]
PowerCLI 12.5 resolved the issue on Powershell 7.2.1 for me https://vdc-download.vmware.com/vmwb-repository/dcr-public/02623a82-abc4-421c-a788-87578bd66208/5ac43430-05f3-489e-af9c-c8de75b65de7/vmware-powercli-125-release-notes.html
